build.gradle.kts

```diff
@@ -72,17 +72,17 @@ subprojects {
 
   // Only apply if the project has the kotlin plugin added:
   plugins.withType<KotlinPluginWrapper> {
-    tasks.withType<KotlinCompile> {
+    tasks.withType<KotlinCompile>().configureEach {
       kotlinOptions {
         jvmTarget = JavaVersion.VERSION_11.toString()
         freeCompilerArgs = listOf("-Xjvm-default=all-compatibility")
       }
       // dependsOn("spotlessKotlinApply")
     }
 
-    tasks.withType<JavaCompile> {
-      sourceCompatibility = JavaVersion.VERSION_11.toString()
-      targetCompatibility = JavaVersion.VERSION_11.toString()
+    configure<JavaPluginExtension> {
+      sourceCompatibility = JavaVersion.VERSION_11
+      targetCompatibility = JavaVersion.VERSION_11
     }
```

Comment:
The change is to configure the `java {}` extension instead, which expresses the intent better and also works against the latest pre-release version of Shadow (9.0.0-beta13).